Clowder auth

mfrdrcks opened the pull request:
This updates the config file and authorization to make use of LDAP or Clowder.

The LDAP is basically unchanged, but Clowder uses a AUTH_URL to send a basic authentication request to. If this isn't the preferred way to authorized to Clowder, then it should be changed now.

mfrdrcks committed:
Added Clowder authentication to the acceptable authentication methods.
Depreciated LDAP_HOSTNAME in favor of AUTH_HOSTNAME (single hostname for all authentications).
Added additional options to config.json.template and README to allow for Clowder authentication.
Have not tested the Clowder SSL authentication, there may still be problems there.

mfrdrcks committed:
Allow for LDAP_HOSTNAME to be specified without AUTH_TYPE to maintain backward comaptibility.

robkooper commented on transformations/auth.py:
```diff
@@ -14,42 +19,69 @@ def login():
+            user = dict()
+        elif re.search(r'Clowder', current_app.config["AUTH_TYPE"], flags=re.IGNORECASE):
+            auth_hostname = current_app.config["AUTH_HOSTNAME"]
+            if "SSL_CERT" not in current_app.config.keys() or current_app.config["SSL_CERT"] is None \
```
why not use if url.startswith("https://") use HTTPSConnection and then if exist SSL_CERT use that.

robkooper replied:
main problem with the shorter and shorter time of certs this will result in a lot of updates.

mfrdrcks replied:
Made this change. Made perfect sense to not require SSL_CERT, and use the system version unless specified.

robkooper commented on transformations/auth.py:
```diff
@@ -14,42 +19,69 @@ def login():
-        except Exception as ex:
-            error = ex
+        user = None
+        if re.search(r'LDAP', current_app.config["AUTH_TYPE"], flags=re.IGNORECASE) or "LDAP_HOSTNAME" \
```
speed wise it might we worth it to use `current_app.config["AUTH_TYPE"].lower() == 'ldap'`

robkooper replied:
if I specify CLOWDER as AUTH_TYPE and specify LDAP_HOSTNAME it will use LDAP.

mfrdrcks replied:
Change to only use LDAP_HOSTNAME as a deciding factor if AUTH_TYPE isn't specified. Changed to not use RE search.

mfrdrcks committed:
Change RE searches to simply do string comparisons.
Change the statement watching for depreciated LDAP_HOSTNAME, to include excluding basing it just on LDAP_HOSTNAME, but also if AUTH_TYPE is specified.
Don't require the SSL_CERT to be specified, but still allow it to be.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/clowder-framework/catalog?pullRequest=24) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/clowder-framework/catalog?pullRequest=24) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/clowder-framework/catalog?pullRequest=24) it.</sub>
